AppHandling/Compile-AppInNavContainer.ps1

```diff
@@ -502,28 +502,43 @@ try {
-
-                                $packageStream = [System.IO.File]::OpenRead($symbolsFile)
-                                $package = [Microsoft.Dynamics.Nav.CodeAnalysis.Packaging.NavAppPackageReader]::Create($PackageStream, $true)
-                                $manifest = $package.ReadNavAppManifest()
-
-                                if ($manifest.application) {
-                                    @{ "publisher" = "Microsoft"; "name" = "Application"; "appId" = ''; "version" = $manifest.Application }
+                                $alToolExe = Join-Path $alcPath 'win32\altool.exe'
+                                $alToolExists = Test-Path -Path $alToolExe -PathType Leaf
+                                if ($alToolExists) {
+                                    $manifest = & "$alToolExe" GetPackageManifest "$symbolsFile" | ConvertFrom-Json
+                                    if ($manifest.PSObject.Properties.Name -eq 'application' -and $manifest.application) {
+                                        @{ "publisher" = "Microsoft"; "name" = "Application"; "appId" = ''; "version" = $manifest.Application }
+                                    }
+                                    if ($manifest.PSObject.Properties.Name -eq 'dependencies') {
+                                        foreach ($dependency in $manifest.dependencies) {
+                                            @{ "publisher" = $dependency.Publisher; "name" = $dependency.name; "appId" = $dependency.id; "Version" = $dependency.Version }
+                                        }
+                                    }
                                 }
-
-                                foreach ($dependency in $manifest.dependencies) {
-                                    $appId = ''
-                                    if ($dependency.psobject.Properties.name -eq 'appid') {
-                                        $appId = $dependency.appid
+                                else {
+                                    Add-Type -Path (Join-Path $alcPath Newtonsoft.Json.dll)
+                                    Add-Type -Path (Join-Path $alcPath System.Collections.Immutable.dll)
+                                    Add-Type -Path (Join-Path $alcPath System.IO.Packaging.dll)
+                                    Add-Type -Path (Join-Path $alcPath Microsoft.Dynamics.Nav.CodeAnalysis.dll)
+
+                                    $packageStream = [System.IO.File]::OpenRead($symbolsFile)
+                                    $package = [Microsoft.Dynamics.Nav.CodeAnalysis.Packaging.NavAppPackageReader]::Create($PackageStream, $true)
+                                    $manifest = $package.ReadNavAppManifest()
+
+                                    if ($manifest.application) {
+                                        @{ "publisher" = "Microsoft"; "name" = "Application"; "appId" = ''; "version" = $manifest.Application }

```

Comment:
The Application's ID is c1335042-3002-4257-bf8a-75c898ccb1b8, if it matters.

Reply:
The Application's ID is not the same for embed partners - they provide their own.
This is why the Application isn't resolved by appid - and uses propagated dependencies